packages/hdwallet-native/src/binance.ts

```diff
@@ -0,0 +1,138 @@
+      return true;
+    }
+
+    async binanceSupportsSecureTransfer(): Promise<boolean> {
+      return false;
+    }
+
+    binanceSupportsNativeShapeShift(): boolean {
+      return false;
+    }
+
+    binanceGetAccountPaths(msg: core.BinanceGetAccountPaths): Array<core.BinanceAccountPath> {
+      return [
+        {
+          addressNList: [0x80000000 + 44, 0x80000000 + 117, 0x80000000 + msg.accountIdx, 0, 0],
+          // hardenedPath: [0x80000000 + 44, 0x80000000 + core.slip44ByCoin(msg.coin), 0x80000000 + msg.accountIdx],
+          // relPath: [0, 0],
+          // description: "Native",
+        },
+      ];
+    }
+
+    binanceNextAccountPath(msg: core.BinanceAccountPath): core.BinanceAccountPath {
+      // Only support one account for now (like portis).
+      return undefined;
+    }
+  };
+}
+
+export function MixinNativeBinanceWallet<TBase extends core.Constructor>(Base: TBase) {
+  return class MixinNativeBinanceWallet extends Base {
+    _supportsBinance = true;
+    #seed = "";
+
+    binanceInitializeWallet(seed: string): void {
+      //get
+      this.#seed = seed;
+    }
+
+    bech32ify(address: ArrayLike<number>, prefix: string): string {
+      const words = toWords(address);
+      return encode(prefix, words);
+    }
+
+    createBinanceAddress(publicKey: string) {
+      const message = SHA256(CryptoJS.enc.Hex.parse(publicKey));
+      const hash = RIPEMD160(message as any).toString();
+      const address = Buffer.from(hash, `hex`);
+      const cosmosAddress = this.bech32ify(address, `bnb`);
+      return cosmosAddress;
+    }
+
+    async binanceGetAddress(msg: core.BinanceGetAddress): Promise<string> {
+      const seed = await mnemonicToSeed(this.#seed);
+
+      const network = getNetwork("bitcoin");
+      const wallet = bitcoin.bip32.fromSeed(seed, network);
+      const path = core.addressNListToBIP32(msg.addressNList);
+      const keypair = await bitcoin.ECPair.fromWIF(wallet.derivePath(path).toWIF(), network);
+      const address = this.createBinanceAddress(keypair.publicKey.toString("hex"));
+
+      return address;
+    }
+
+    async binanceSignTx(msg: BinanceSignTx, mnemonic: string, xpriv: string, from: string): Promise<BinanceSignedTx> {
+      const seed = await mnemonicToSeed(this.#seed);
+
+      const network = getNetwork("cosmos");
+      const hdkey = bitcoin.bip32.fromSeed(seed, network);
+      const path = core.addressNListToBIP32(msg.addressNList);
+
+      let keypair = await bitcoin.ECPair.fromWIF(hdkey.derivePath(path).toWIF(), network);
+      let privateKey = keypair.privateKey.toString("hex");
+
+      //use sdk to build Amino encoded hex transaction
+      const client = new BncClient("https://dex.binance.org"); //broadcast not used but available
+      await client.chooseNetwork("mainnet");
+      await client.setPrivateKey(privateKey);
+      await client.initChain();
+
+      //let fromAddress = msg
+      const addressFrom = msg.tx.msgs[0].inputs[0].address;
```

Comment:
Assuming only ever 1 input address and 1 output address. Maybe add a todo to handle multiple if that can happen on bnb?